lib/kubernetes-deploy/kubernetes_resource/pod.rb

```diff
@@ -19,6 +19,7 @@ def sync(pod_data = nil)
       if pod_data.blank?
         raw_json, _err, st = kubectl.run("get", type, @name, "-a", "--output=json")
         pod_data = JSON.parse(raw_json) if st.success?
+        raise_predates_deploy_error if pod_data.present? && unmanaged? && !@deploy_started
```

Comment:
As discussed IRL, `deploy_started` should probably have been named `deploy_started_at`: it is a timestamp indicating when _this specific resource_ was deployed (or it is nil when it hasn't been deployed yet). What we're doing here is checking whether an unmanaged pod existed before the current deploy created it.